vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -210,9 +196,6 @@ - (void)dealloc
   [NSNotificationCenter.defaultCenter removeObserver:self];
   [self.notificationStartupTimer invalidate];
 
-  if (rewardsDatabase) {
-    databaseQueue->DeleteSoon(FROM_HERE, rewardsDatabase);
-  }
```

Comment:
No manual cleanup required on deconstructor?

Reply:
When the `Remote` is destroyed, it's pipe is destroying and the implementation on the other end of the pipe is destroyed (via `mojo::MakeSelfOwnedReceiver`).